### PR TITLE
Add Form::SubmitComponent

### PR DESCRIPTION
**Pull request by Spone**
Related to #4

### CODE REVIEW

File: app/components/view_component/form/submit_component.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+module ViewComponent
+  module Form
+    class SubmitComponent < BaseComponent
+      attr_reader :value
+
+      def initialize(form, value, options = {})
+        @value = value
+        @options = options
+
+        super(form, nil, options)
```
Comment (Spone) [inline app/components/view_component/form/submit_component.rb:12]: Not sure about passing `nil` here...

File: lib/view_component/form/builder.rb [hunk 2]
```diff
@@ -52,7 +52,7 @@ def submit(value = nil, options = {})
           value = nil
         end
         value ||= submit_default_value
-        render_component(:submit, self, @object_name, value, options)
```
Comment (Spone) [inline lib/view_component/form/builder.rb:55]: Not sure about removing `@object_name` here

File: app/components/view_component/form/submit_component.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module ViewComponent
+  module Form
+    class SubmitComponent < BaseComponent
+      attr_reader :value
+
+      def initialize(form, value, options = {})
+        @value = value
+        @options = options
```
Comment (nicolas-brousse) [inline app/components/view_component/form/submit_component.rb:10]: Not necessary, it's already done by BaseComponent.

File: app/components/view_component/form/submit_component.rb (hunk 1 above)
Comment (nicolas-brousse) [reply app/components/view_component/form/submit_component.rb:12]: Sounds okay

File: lib/view_component/form/builder.rb (hunk 2 above)
Comment (nicolas-brousse) [reply lib/view_component/form/builder.rb:55]: Sounds okay

File: spec/view_component/form/submit_component_spec.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+RSpec.describe ViewComponent::Form::SubmitComponent, type: :component do
+  let(:object)  { OpenStruct.new }
+  let(:form)    { form_with(object) }
+  let(:options) { {} }
+
+  let(:component) { render_inline(described_class.new(form, "", options)) }
```
Comment (nicolas-brousse) [inline spec/view_component/form/submit_component_spec.rb:8]:
```suggestion
  let(:value) { "" }

  let(:component) { render_inline(described_class.new(form, value, options)) }
```

Test with a custom value may be nice

**Review by nicolas-brousse — CHANGES_REQUESTED** [review]
LGTM, I just add some suggestions

**Review by nicolas-brousse — APPROVED** [review]

**Pull request merged**